libpod/runtime_ctr.go

```diff
@@ -76,7 +76,11 @@ func (r *Runtime) NewContainer(ctx context.Context, rSpec *spec.Spec, options ..
 		ctr.config.LogPath = filepath.Join(ctr.config.StaticDir, "ctr.log")
 	}
 	if ctr.config.ShmDir == "" {
-		ctr.config.ShmDir = filepath.Join(ctr.bundlePath(), "shm")
+		if ctr.config.UserNSRoot == "" {
+			ctr.config.ShmDir = filepath.Join(ctr.bundlePath(), "shm")
```

Comment:
Should we change bundlePath() to return UserNSRoot if we are running in a user namespace?

Reply:
I think the userNS path should be used only to setup the bind mounts and by runc, everything else should be done in the same way as before